### ISSUE TITLE
The implementation of grad clipping is not correct when gradient accumulation is enabled

### ISSUE THREAD

**Issue by yangyiben** [huggingface/transformers#1563]
torch.nn.utils.clip_grad_norm_ should be applied before optimizer.step() not after each backward

## 🐛 Bug

Model I am using (Bert, XLNet....):

Language I am using the model on (English, Chinese....):

The problem arise when using:
* [ ] the official example scripts: (give details)
* [ ] my own modified scripts: (give details)

The tasks I am working on is:
* [ ] an official GLUE/SQUaD task: (give the name)
* [ ] my own task or dataset: (give details)

## To Reproduce

1.
2.
3.

## Environment

* OS:
* Python version:
* PyTorch version:
* PyTorch Transformers version (or branch):
* Using GPU ?
* Distributed of parallel setup ?
* Any other relevant information:

**Comment by BramVanroy** [comment]
This should be fixed in https://github.com/huggingface/transformers/pull/1580

**Comment by pminervini** [comment]
yeah @yangyiben please let me know if that merge fixes it

**Comment by thomwolf** [comment]
#1580 is now merged
